client/src/App.tsx

```diff
@@ -320,18 +326,24 @@ function App() {
     let _project = (project == 'mathlib-demo' ? null : project)
     if (code === contentFromUrl) {
       if (url !== null) {
-        let args = {project: _project, url: encodeURIComponent(url), code: null}
+        let args = {project: _project, url: encodeURIComponent(url), code: null, codez: null}
         history.replaceState(undefined, undefined!, formatArgs(args))
       } else {
-        let args = {project: _project, url: null, code: null}
+        let args = {project: _project, url: null, code: null, codez: null}
         history.replaceState(undefined, undefined!, formatArgs(args))
       }
     } else if (code === "") {
-      let args = {project: _project, url: null, code: null}
+      let args = {project: _project, url: null, code: null, codez: null}
       history.replaceState(undefined, undefined!, formatArgs(args))
     } else {
-      let args = {project: _project, url: null, code: fixedEncodeURIComponent(code)}
-      history.replaceState(undefined, undefined!, formatArgs(args))
+      const compressed = LZString.compressToBase64(code).replace(/=*$/, '');
+      if(compressed.length < code.length) {
```

Comment:
I made this only use compression if its actually a win (see charts in PR description).

For very small character counts it isn't always a win, although it also doesn't really matter anyway.
If you'd like it to always format URLs with codez I can remove this check.

Reply:
Fixed in https://github.com/leanprover-community/lean4web/commit/fa4342d69389157bc13da96127196212e14604c1